src/commons/temporal.c

```diff
@@ -47,3 +48,77 @@ char *temporal_get_string_time(const char* format) {
 
 	return str_time;
 }
+
+t_temporal* temporal_create(void) {
+    t_temporal* self = malloc(sizeof(t_temporal));
+    struct timespec* current = malloc(sizeof(struct timespec));
+    
+    self -> elapsed_ms = 0;
+    self -> state = RUNNING;
+    self -> current = current;
+    
+    clock_gettime(CLOCK_MONOTONIC_RAW, current);
+    
+    return self;
+}
+
+void temporal_destroy(t_temporal* temporal) {
+    if (temporal -> current) {
+		free(temporal -> current);
+	}
+
+    free(temporal);
+}
+
+int64_t temporal_gettime(t_temporal* temporal) {
+    if (temporal -> state == STOPPED) {
+        return temporal -> elapsed_ms;
+    }
+
+    t_temporal* now = temporal_create();
+
+    int64_t delta_ms = (now -> current -> tv_sec - temporal -> current -> tv_sec) * 1000 
+                        + (now -> current -> tv_nsec - temporal -> current -> tv_nsec) / 1000000;
+    
+    temporal_destroy(now);
+
+    return delta_ms + temporal -> elapsed_ms;
+}
+
+int64_t temporal_gettime_since_running(t_temporal* temporal) {
+	if (temporal -> state == STOPPED) {
+        return 0;
+    }
+
+    t_temporal* now = temporal_create();
+
+    int64_t delta_ms = (now -> current -> tv_sec - temporal -> current -> tv_sec) * 1000 
+                        + (now -> current -> tv_nsec - temporal -> current -> tv_nsec) / 1000000;
+    
+    temporal_destroy(now);
+
+    return delta_ms;
+}
+
+void temporal_stop(t_temporal* temporal) {
+    temporal -> elapsed_ms += temporal_gettime_since_running(temporal);
```

Comment:
¿`temporal_gettime_since_running()` serviría exclusivamente para este caso o ves que sea útil (por ejemplo, para algo del TP anterior)?

Sino en su lugar creo que podríamos dejar algo como:
```suggestion
	if (temporal->state == STOPPED) {
		return;
	}
	temporal -> elapsed_ms += temporal_gettime(temporal);
```

Reply:
Me gustaría decirte que tiene alguna utilidad para algo más, pero no se me ocurre algo que no sea sumamente rebuscado jajajajaj, creé la función como auxiliar para este caso puntual, si te parece, la dejamos como función privada, o directamente la sacamos.

Reply:
Genial, lo sacamos, lo que sí, en lugar de usar `temporal_gettime()`, agrego una función privada para evitar la lógica repetida en `temporal_gettime()`, ya que, a pesar de que es muy parecido, al usar `temporal_gettime()` se le va a adicionar todo el tiempo transcurrido (`delta_ms + temporal->elapsed`), en lugar de el último período (`delta_ms`), entonces, creo una función para el cálculo de `delta_ms`, y la uso tanto en `temporal_gettime()` como en `temporal_stop()`.